src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -163,21 +233,27 @@ public override void VisitModuleDeclarationSyntax(ModuleDeclarationSyntax syntax
 
                 var declaredType = syntax.GetDeclaredType(this.binder.TargetScope, moduleSemanticModel);
 
-                this.ValidateDecortors(syntax.Decorators, declaredType, diagnostics);
+                this.ValidateDecorators(syntax.Decorators, declaredType, diagnostics);
 
                 if (moduleSemanticModel.HasErrors())
                 {
                     diagnostics.Write(DiagnosticBuilder.ForPosition(syntax.Path).ReferencedModuleHasErrors());
                 }
 
+                if (syntax.Value is ForSyntax)
+                {
+                    // TODO: Narrowing doesn't work

```

Comment:
Mind adding a bit more detail to this comment?

Reply:
I will remove it. This was actually fixed.